src/Tuura/Concept/STG/Translation.hs

```diff
@@ -86,9 +86,10 @@ initVal s ls = sum (map (\x -> if (fst x == s) then fromEnum (snd x) else 0) ls)
 readArc :: String -> String -> [String]
 readArc f t = [f ++ " " ++ t, t ++ " " ++ f]
 
-genInvStrs :: (Eq a, Show a) => [[Transition a]] -> [String]
-genInvStrs invs
-        | invs      == [] = []
-        | otherwise = info ++ map (\i -> unwords (map show i)) invs
+genInvStrs :: (Ord a, Show a) => Invariant (Transition a) -> String
+genInvStrs (NeverAll es)
```

Comment:
Again, get rid of explicit recursion -- it's almost never required in Haskell because there are so many powerful combinators. In this case you most probably need to first `map format es` and then do something like `intercalate "&&"`.